packages/calcite-components/src/components/tabs/tabs.tsx

```diff
@@ -140,19 +68,35 @@ export class Tabs {
 
   @Element() el: HTMLCalciteTabsElement;
 
+  private defaultSlotChangeHandler = (event): void => {
+    this.tabs = slotChangeGetAssignedElements(event, "calcite-tab") as HTMLCalciteTabElement[];
+  };
+
+  private slotEl: HTMLSlotElement;
+
   /**
    *
    * Stores an array of ids of `<calcite-tab-titles>`s to match up ARIA
    * attributes.
    */
   @State() titles: HTMLCalciteTabTitleElement[] = [];
 
+  @Watch("titles")
+  titlesWatcher(): void {
+    this.registryHandler();
+  }
+
   /**
    *
    * Stores an array of ids of `<calcite-tab>`s to match up ARIA attributes.
    */
   @State() tabs: HTMLCalciteTabElement[] = [];
 
+  @Watch("tabs")
+  tabsWatcher(): void {
+    this.registryHandler();
```

Comment:
You read my mind, I was thinking of renaming that function, but decided not to at first to minimize the diff, but I do agree this would be nice to rename it more semantically.  Will do :)

Reply:
Took a stab at a name: `updateAriaSettings`.